test/interactive/command.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Helper functions for running interactive CLI sessions from Go
+package interactive
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// could be public so extensions of this can test themselves
+var defaultRun func(*exec.Cmd) error
+
+func init() {
+	defaultRun = func(c *exec.Cmd) error { return c.Run() }
+}
+
+// Command represents running an executable, to make it easy to write interactive "shell scripts" in Go.
+// This routes standard in, out, and error from and to the calling program, intended usually to be a login shell run by a person.
+type Command struct {
```

Comment:
I feel the functionality is a bit overlapping with https://github.com/knative/pkg/blob/master/test/cmd/command.go, any chance we can move it there?

Reply:
That library explicitly runs commands in a different environment than the caller's environment, while this library is intended for commands running in the same environment (like a shell script). Given it's just a tiny wrapper around exec.Command(), there's nothing else to combine.

Reply:
SG